bastion/src/envelope.rs

```diff
@@ -0,0 +1,190 @@
+use crate::broadcast::Sender;
+use crate::message::{BastionMessage, Message, Msg};
+use crate::path::BastionPath;
+use crate::system::SYSTEM;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub(crate) struct Envelope {
+    pub(crate) msg: BastionMessage,
+    pub(crate) sign: RefAddr,
+}
+
+#[derive(Debug)]
+/// A struct containing `Msg` and its `RefAddr`.
+pub struct SignedMessage {
```

Comment:
Should this replace `Msg`?

Reply:
That was my first thought but BastionMessage::Start,Stop,etc also need signatures.